core/templates/dev/head/pages/story-editor-page/editor-tab/story-node-editor.directive.ts

```diff
@@ -134,6 +152,7 @@ angular.module('oppia').directive('storyNodeEditor', [
             StoryUpdateService.setStoryNodeExplorationId(
               $scope.story, $scope.getId(), explorationId);
             $scope.currentExplorationId = explorationId;
+            $scope.expIdIsValid = _isExpIdValid($scope.currentExplorationId);
```

Comment:
If the expId is invalid then why we are going to update the storynode with invalidId on line 152?

Reply:
Ok. Story node gets updated only if the expId is valid